### DISCUSSION TITLE
V4 - ?blackpill? - diy

### DISCUSSION THREAD

**Discussion by 5chufti** [keirf/greaseweazle#113] [Q&A] [answered]
Hi,
please excuse my (maybe stupid) question:
what functional difference is there between the F1 (bluepill) and V4 version?
would the V4 Artery AT32F415 based fw run on an ST32F411 (aka blackpill) giving an equivalent to the F1 version?
I'm not aware of the detailed differences between AT32F415 and ST32F411 but I assume the similarity is not accidental?

**Comment by keirf** [comment, marked as answer]
The V4 runs on AT32F403 (although actually AT32F415 is also supported). These are actually more like STM32F1xx chips, but with a Cortex-M4 core. They are not so compatible with STM32F4xx series. Actually the AT32F403 is a pretty close clone of the STM32F103 as used on Blue Pill. You could probably drop one in as a replacement on a Blue Pill! I have used them in V4 because of availability: Try getting a reliable source of genuine STM32 chips right now! 

Functionality differences: V4 has buffered outputs and better assignment of MCU pins means more pins connected to the floppy interface (eg pins 2,4,6 are user-definable outputs; pin 34 is connected as a user-readable input). Jumperless update, support bussed floppy drives, flippy drive support, ... A lot of this is actually enabled by the V4 PCB rather than being because the AT32F403 is intrinsically better than or even different than the STM32F103.

Hope that clears it up!

**Reply by 5chufti** [reply]
Hi, thanks for the quick reply. So if I take my known good genuine STM32F103 bluepill I can wire it up according to V4 schematic (including buffers), flash V4 firmware and it should behave like V4?

**Reply by keirf** [reply]
Unfortunately not, since AT32F403 has features beyond STM32F103 (more RAM, Cortex-M4 which can be clocked much faster, 32-bit timers) which, since I do a separate build, the firmware can rely on.

**Reply by keirf** [reply]
I could possibly support F103 on V4 layout in the existing F1 firmware build though. It's not really that interesting for a one-off experiment though.

**Reply by 5chufti** [reply]
ok, so coming back to my initially intended question: is there any fw functionality in the V4 (due to AT32F403 features) that one would miss on the F1 (seeing that the F1 plus mini has the additional user pins too)?

**Reply by keirf** [reply]
This is covered on the Models page: https://github.com/keirf/Greaseweazle/wiki/Greaseweazle-Models
They are identical in features (12v power feature is a board feature and not firmware related at all).

**Reply by keirf** [reply]
So F1 Plus schematic is a good basis for anyone with good STM32F103 chips. They're just hard to find in quantity!

**Reply by 5chufti** [reply]
Hi, thanks again for your quick answer. I did see the models page but for me it only covered hw differences and I wanted to avoid starting with (soon to be) outdated hw and loosing access to new fw features not supported on old hw platform.
Unfortunately I was too late to buy on ebay and usually "selected shops in eu" charge prohibitive shipping cost to Austria, so I will go the DIY path.

**Reply by keirf** [reply]
I have no immediate plans to outdate the V4 nor the F1 Plus. I have hundreds of V4s to sell!
